corehq/apps/accounting/models.py

```diff
@@ -500,13 +507,24 @@ def _get_account_by_created_by_domain(cls, domain):
     def get_enterprise_restricted_signup_accounts(cls):
         return BillingAccount.objects.filter(is_customer_billing_account=True, restrict_signup=True)
 
+    @classmethod
+    def get_enterprise_permissions_domains(cls, domain):
+        account = cls.get_account_by_domain(domain)
+        if account and account.permissions_source_domain:
+            return list(set(account.get_domains()) - set(account.permissions_ignore_domains))
+        return []
+
     @property
     def autopay_card(self):
         if not self.auto_pay_enabled:
             return None
 
         return StripePaymentMethod.objects.get(web_user=self.auto_pay_user).get_autopay_card(self)
 
+    def get_domains(self):
+        subscriptions = Subscription.visible_objects.filter(account_id=self.id, is_active=True)
+        return [s.subscriber.domain for s in subscriptions]
```

Comment:
This would be more efficient as a single query:
```suggestion
        return list(Subscription.visible_objects.filter(account_id=self.id, is_active=True).values_list('subscriber__domain', flat=True))
```